contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,661 @@
+        bytes32 _dependencyTypeId,
+        uint256 _index
+    )
+        external
+        onlyAdminACL(this.removeDependencyTypeAdditionalCDNAtIndex.selector)
+        onlyExistingDependencyType(_dependencyTypeId)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        require(_index < additionalCDNCount, "Asset index out of range");
+
+        uint24 lastElementIndex = additionalCDNCount - 1;
+
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+                _index
+            ] = dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            lastElementIndex
+        ];
+        delete dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            lastElementIndex
+        ];
+
+        dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount = lastElementIndex;
+
+        emit DependencyTypeAdditionalCDNRemoved(_dependencyTypeId, _index);
+    }
+
+    function updateDependencyTypeAdditionalCDNAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index,
+        string memory _additionalCDN
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypeAdditionalCDNAtIndex.selector)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        require(_index < additionalCDNCount, "Asset index out of range");
+
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            _index
+        ] = _additionalCDN;
+
+        emit DependencyTypeAdditionalCDNUpdated(
+            _dependencyTypeId,
+            _additionalCDN,
+            _index
+        );
+    }
+
+    function addDependencyTypeAdditionalRepository(
+        bytes32 _dependencyTypeId,
+        string memory _additionalRepository
+    )
+        external
+        onlyAdminACL(this.addDependencyTypeAdditionalRepository.selector)
+        onlyNonEmptyString(_additionalRepository)
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount;
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+                additionalRepositoryCount
+            ] = _additionalRepository;
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositoryCount =
+            additionalRepositoryCount +
+            1;
+
+        emit DependencyTypeAdditionalRepositoryUpdated(
+            _dependencyTypeId,
+            _additionalRepository,
+            additionalRepositoryCount
+        );
+    }
+
+    function removeDependencyTypeAdditionalRepositoryAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index
+    )
+        external
+        onlyAdminACL(
+            this.removeDependencyTypeAdditionalRepositoryAtIndex.selector
+        )
+        onlyExistingDependencyType(_dependencyTypeId)
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
```

Comment:
Following a similar pattern to the flex contract here so that we never have a sparse "array". I don't think order should ever matter here.